core/primitives/src/state_record.rs

```diff
@@ -163,3 +163,7 @@ pub fn state_record_to_account_id(state_record: &StateRecord) -> &AccountId {
 pub fn is_contract_code_key(key: &[u8]) -> bool {
     key[0] == col::CONTRACT_CODE
 }
+
+pub fn is_delayed_receipt_key(key: &[u8]) -> bool {
```

Comment:
I believe that this will panic if we pass `&vec![]`.  Is that fine or should we add a check for the length here?

Reply:
I think it is fine, because it should never be called for incorrect (e.g. empty) keys, and if it happened, then protocol had been broken. We also use raw trie keys in other places of the code, so I didn't reuse `TrieKey` logic here.

Reply:
I see... makes sense.  Should we add an `assert` though to help indicate to the reader that this is a precondition for this function.